storage/myisam/ha_myisam.cc

```diff
@@ -398,12 +444,79 @@ int table2myisam(TABLE *table_arg, MI_KEYDEF **keydef_out,
     DBUG_PRINT("loop", ("length: %d  type: %d",
                         recinfo_pos[-1].length,recinfo_pos[-1].type));
   }
+   for(uint i=0;i<share->uniques;i++)
+ {
+	  (*recinfo_pos).type=   FIELD_CHECK;
+     (*recinfo_pos).length= MI_UNIQUE_HASH_LENGTH;
+	   recinfo_pos++;
+	   share->reclength+= MI_UNIQUE_HASH_LENGTH;
+       share->stored_rec_length+= MI_UNIQUE_HASH_LENGTH;
+  }
+
   *records_out= (uint) (recinfo_pos - recinfo);
   DBUG_RETURN(0);
 }
 
 
 /*
+compares key segments.
```

Comment:
Again function description does not really tell me what happens here. What is key segment ?